fairyfishgui.py

```diff
@@ -158,6 +162,31 @@ def char_board(self):
             lastchar = c
         return board
 
+    def update_pockets(self):
+        for pocket in self.pockets:		# clear all the pocket pieces; remain the keys
+            for piece in self.pockets[pocket]:
+                self.pockets[pocket][piece] = 0
+        white_pocket = []
+        black_pocket = []
+        pocket_pieces = re.findall(r"\[(.*)\]",self.fen())
+        if pocket_pieces:
+            prefix = ''
+            for i in pocket_pieces[0]:
+                if i == '+':
```

Comment:
Having promoted pieces in hand is not supported as they are always either demoted or fully converted to the promoted type (depending on dropLoop), so if that is what this code is referring to it can be simplified away.